Sources/LeafKit/LeafLexer/LeafParameterTypes.swift

```diff
@@ -50,30 +50,111 @@ public enum Parameter: Equatable, CustomStringConvertible {
+    internal var bool: Bool? {
+        guard isBooleanValued else { return nil }
+        return [.true, .yes].contains(self)
+    }
+}
 
-    case plus = "+"
-    case minus = "-"
-    case divide = "/"
-    case multiply = "*"
+// MARK: - Operator Symbols
 
-    case and = "&&"
-    case or = "||"
+/// Mathematical and Logical operators
+public enum LeafOperator: String, Equatable, CustomStringConvertible, CaseIterable {
+    // MARK: Public - Cases
+    
+    // Operator types:              Logic      Exist.       UnPre        Scope
+    //                                |   Math    |   Infix   |   UnPost   |
+    //   Logical Tests          --------------------------------------------
+    case not = "!"              //    X                       X
+    case equal = "=="           //    X                 X
+    case unequal = "!="         //    X                 X
+    case greater = ">"          //    X                 X
+    case greaterOrEqual = ">="  //    X                 X
+    case lesser = "<"           //    X                 X
+    case lesserOrEqual = "<="   //    X                 X
+    case and = "&&"             //    X                 X
+    case or = "||"              //    X                 X
+    //   Mathematical Calcs     // -----------------------------------------
+    case plus = "+"             //          X           X
+    case minus = "-"            //          X     X     X     X
+    case divide = "/"           //          X           X
+    case multiply = "*"         //          X           X
+    case modulo = "%"           //          X           X
+    //   Assignment/Existential //
+    case assignment = "="       //                X     X
+    case nilCoalesce = "??"     //                X     X
+    case evaluate = "`"         //                X           X
+    //   Scoping
+    case scopeRoot = "$"        //                            X           X
+    case scopeMember = "."      //                      X                 X
+    case subOpen = "["          //                      X                 X
+    case subClose = "]"         //                                  X     X
     
     /// Raw string value of the operator - eg `!=`
     public var description: String { return rawValue }
+    
+    // MARK: Internal Only
+    
+    // State booleans
+    internal var logical: Bool { Self.states["logical"]!.contains(self) }
+    internal var mathematical: Bool { Self.states["mathematical"]!.contains(self) }
+    internal var existential: Bool { Self.states["existential"]!.contains(self) }
+    internal var scoping: Bool { Self.states["scoping"]!.contains(self) }
+    
+    internal var unaryPrefix: Bool { Self.states["unaryPrefix"]!.contains(self) }
+    internal var unaryPostfix: Bool { Self.states["unaryPostfix"]!.contains(self) }
+    internal var infix: Bool { Self.states["unaryPostfix"]!.contains(self) }
+    
+    internal var available: Bool { !Self.states["unavailable"]!.contains(self) }
```

Comment:
State identities of operators for convenience in referring to them during lex/parse/resolving